examples/decomposition/topics_extraction_with_online_lda.py

```diff
@@ -0,0 +1,97 @@
+"""
+==================================================
+Topics extraction with Latent Dirichlet Allocation
+==================================================
+
+This is an example of showing how to extract topics from
+20 newsgroup dataset with Latent Dirichlet Allocation.
+It uses both online and batch update methods to train LDA
+model and shows that online method can get meaningful topics
+with less iterations.
+
+The code is modified from the
+"Topics extraction with Non-Negative Matrix Factorization"
+example.
+
+"""
+
+# Authors: Chyi-Kwei Yau
+
+from __future__ import print_function
+from time import time
+from sklearn.datasets import fetch_20newsgroups
+from sklearn.feature_extraction.text import CountVectorizer
+from sklearn.decomposition import LatentDirichletAllocation
+
+
+def print_top_words(model, feature_names, n_top_words):
+    for topic_idx, topic in enumerate(model.components_):
+        print("Topic #%d:" % topic_idx)
+        print(" ".join([feature_names[i] for i in topic.argsort()[:-n_top_words - 1:-1]]))
+    print()
+
+
+def lda_example():
```

Comment:
I'd rather not have a function + call in __main__ but as a script, as in the other examples.